tomcat-gcp-lib/src/main/java/com/google/cloud/runtimes/tomcat/session/DatastoreManager.java

```diff
@@ -0,0 +1,212 @@
+ * with {@link DatastoreStore}.<br/>
+ * Example configuration:</p>
+ *
+ * <pre>
+ *   {@code
+ *   <Valve className="com.google.cloud.runtimes.tomcat.session.DatastoreValve" />
+ *   <Manager className="com.google.cloud.runtimes.tomcat.session.DatastoreManager" >
+ *     <Store className="com.google.cloud.runtimes.tomcat.session.DatastoreStore" />
+ *   </Manager>
+ *   }
+ * </pre>
+ *
+ * <p>The session is never stored locally and always fetch from the datastore.</p>
+ */
+public class DatastoreManager extends ManagerBase implements StoreManager {
+
+  private static final Log log = LogFactory.getLog(DatastoreManager.class);
+
+  private static final String name = "DatastoreManager";
+
+  /**
+   * The store will be in charge of all the interaction with the Datastore.
+   */
+  protected Store store = null;
+
+  /**
+   * {@inheritDoc}
+   *
+   * <p>Ensure that a store is present and initialized.</p>
+   *
+   * @throws LifecycleException If an error occur during the store initialisation
+   */
+  @Override
+  protected synchronized void startInternal() throws LifecycleException {
+    super.startInternal();
+
+    if (store == null) {
+      throw new LifecycleException("No Store configured, persistence disabled");
+    } else if (store instanceof Lifecycle) {
+      ((Lifecycle) store).start();
+    }
+
+    setState(LifecycleState.STARTING);
+  }
+
+  /**
+   * Search into the store for an existing session with the specified id.
+   *
+   * @param id The session id for the session to be returned
+   * @return The request session or null if a session with the requested ID could not be found
+   * @throws IOException If an input/output error occurs while processing this request
+   */
+  @Override
+  public Session findSession(String id) throws IOException {
+    log.debug("Datastore manager is loading session: " + id);
+    Session session = null;
+
+    try {
+      session = this.getStore().load(id);
+    } catch (ClassNotFoundException ex) {
+      log.warn("An error occurred during session deserialization" + ex);
+    }
+
+    return session;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * <p>Note: Sessions are loaded at each request therefore no session is loaded at
+   * initialization.</p>
+   *
+   * @throws ClassNotFoundException Cannot occurs
+   * @throws IOException Cannot occurs
+   */
+  @Override
+  public void load() throws ClassNotFoundException, IOException {}
+
+  /**
+   * {@inheritDoc}
+   *
+   * <p>Note: Sessions are persisted after each requests but never saved into the local manager,
+   * therefore no operation is needed during unload.</p>
+   *
+   * @throws IOException Cannot occurs
+   */
+  @Override
+  public void unload() throws IOException {}
+
+  /**
+   * Remove the Session from the manager but not from the Datastore
+   *
+   * @param session The session to remove.
+   */
+  @Override
+  public void removeSuper(Session session) {
```

Comment:
removeSuper is also used in the Valve in order to remove the session locally without removing it from the Datastore.